tests/test_convertReferenceCatalog.py

```diff
@@ -183,6 +256,127 @@ def testValidateParallax(self):
+        self.config.ra_err_name = 'ra_error'
+        self.config.dec_err_name = 'dec_error'
+        self.config.pm_ra_name = 'pmra'
+        self.config.pm_dec_name = 'pmdec'
+        self.config.pm_ra_err_name = 'pmra_error'
+        self.config.pm_dec_err_name = 'pmdec_error'
+        self.config.parallax_name = 'parallax'
+        self.config.parallax_err_name = 'parallax_error'
+        self.config.epoch_name = 'unixtime'
+        self.config.epoch_format = 'unix'
+        self.config.epoch_scale = 'tai'
+        self.depth = 2  # very small depth, for as few pixels as possible.
+        self.indexer = HtmIndexer(self.depth)
+        self.htm = lsst.sphgeom.HtmPixelization(self.depth)
+        converter = ConvertReferenceCatalogTask(output_dir=tempPath, config=self.config)
+        dtype = [('id', '<f8'), ('ra', '<f8'), ('dec', '<f8'), ('ra_err', '<f8'), ('dec_err', '<f8'),
+                 ('a', '<f8'), ('a_err', '<f8')]
+        self.schema, self.key_map = converter.makeSchema(dtype)
+        self.fileReader = ReadTextCatalogTask()
+
+        self.fakeInput = self.makeSkyCatalog(outPath=None, size=5, idStart=6543)
+        self.matchedPixels = np.array([1, 1, 2, 2, 3])
+        self.tempDir2 = tempfile.TemporaryDirectory()
+        tempPath = self.tempDir2.name
+        self.filenames = {x: os.path.join(tempPath, "%d.fits" % x) for x in set(self.matchedPixels)}
+
+        self.worker = ConvertGaiaManager(self.filenames,
+                                         self.config,
+                                         self.fileReader,
+                                         self.indexer,
+                                         self.schema,
+                                         self.key_map,
+                                         self.htm.universe()[0],
+                                         addRefCatMetadata,
+                                         self.log)
+
+    def tearDown(self):
+        self.tempDir.cleanup()
+        self.tempDir2.cleanup()
+
+    def test_positionSetting(self):
+        """Test the _setProperMotion, _setParallax, and
+        _setCoordinateCovariance methods.
+        """
+        outputCatalog = SimpleCatalog(self.worker.schema)
+        outputCatalog.resize(len(self.fakeInput))
+
+        # Set coordinate errors and covariances:
+        coordErr = self.worker._getCoordErr(self.fakeInput)
+        for name, array in coordErr.items():
+            outputCatalog[name] = array
+
+        for outputRow, inputRow in zip(outputCatalog, self.fakeInput):
+            self.worker._setProperMotion(outputRow, inputRow)
+            self.worker._setParallax(outputRow, inputRow)
+            self.worker._setCoordinateCovariance(outputRow, inputRow)
+
+        coordConvert = (self.worker.coord_err_unit).to(u.radian)
+        pmConvert = (self.worker.config.pm_scale * u.milliarcsecond).to_value(u.radian)
+        parallaxConvert = (self.config.parallax_scale * u.milliarcsecond).to_value(u.radian)
```

Comment:
I changed them to all be `self.worker.config`. As written, this doesn't fail because `self.worker.config` is `self.config`, but obviously it ought to be consistent.